### DISCUSSION TITLE
Programmatically Extracting g-function/lntts Data

### DISCUSSION THREAD

**Discussion by mitchute** [j-c-cook/ghedt#95] [General]
What's the best way to extract g-function (STS and LTS) and ln(t/ts) values for a particular solution of the bisection search?

For example, I have this:

```python
    # Single U-tube
    # -------------
    # Perform field selection using bisection search
    coordinates_domain = dt.domains.square_and_near_square(1, 64, bh_spacing)

    bisection_search = dt.search_routines.Bisection1D(coordinates_domain, vol_flow_rate, borehole, single_u_tube,
                                                      fluid, pipe_single, grout, soil, sim_params,
                                                      loads, disp=False)

    # Find the near-square design for a single U-tube and size it.
    ghe = bisection_search.ghe
    ghe.compute_g_functions()
```

I see that the `ghe` variable (which looks like it is an instance of the `GFunction` class) has `self.g_lts` and `self.log_time` member variables. And the `BaseGHE` class has `combine_sts_lts` and `grab_g_function` methods.

What do you recommend?

**Comment by j-c-cook** [comment]
Check out how I exported the g-functions in this `oak_ridge_export` function. I limit the number of short time step g-function values and sample it because either OpenStudio or EnergyPlus didn't like all of them.  

https://github.com/j-c-cook/ghedt/blob/3a4ad151163bb9a8aec2aedb6f0e4215cb7c41bd/ghedt/search_routines.py#L385-L405
